src/dsp_tools/utils/xmlupload/upload_stashed_resptr_props.py

```diff
@@ -97,6 +74,125 @@ def upload_stashed_resptr_props(
     return nonapplied_resptr_props
 
 
+def _upload_all_resptr_props_of_single_resource(
+    resource_in_triplestore: dict[str, Any],
+    link_prop: XMLProperty,
+    res_iri: str,
+    stashed_resource: XMLResource,
+    resptrs: list[str],
+    id2iri_mapping: dict[str, str],
+    con: Connection,
+    nonapplied_resptr_props: dict[XMLResource, dict[XMLProperty, list[str]]],
+    verbose: bool,
+) -> dict[XMLResource, dict[XMLProperty, list[str]]]:
+    """
+    This function takes one resource stashed resource and resptr-props that are specific to a property.
+    It uploads them to the triplestore and removes them from the nonapplied_resptr_props dictionary.
+
+    Args:
+        resource_in_triplestore: The resource retried from the triplestore
+        link_prop: the link property object
+        res_iri: the IRI as given by the DSP-API
+        stashed_resource: The resource from the stash
+        resptrs: List with all the resptr-props for that one property
+        id2iri_mapping: mapping of internal id to the IRI from the DSP-API
+        con: Connection to the DSP-API
+        nonapplied_resptr_props: stashed resources
+        verbose: If True, more information is logged
+
+    Returns:
+        The nonapplied_resptr_props dictionary with the uploaded resource removed
+    """
+    for resptr in resptrs.copy():
+        jsondata = _create_resptr_prop_json_object_to_update(
+            stashed_resource=stashed_resource,
+            resource_in_triplestore=resource_in_triplestore,
+            property_name=resptr,
+            link_prop=link_prop,
+            res_iri=res_iri,
+            id2iri_mapping=id2iri_mapping,
+        )
+        try:
+            try_network_action(con.post, route="/v2/values", jsondata=jsondata)
+        except BaseError as err:
+            # print the message to keep track of the cause for the failure.
+            # Apart from that, no action is necessary:
+            # this resource will remain in nonapplied_resptr_props, which will be handled by the caller
+            orig_err_msg = err.orig_err_msg_from_api or err.message
+            err_msg = f"Unable to upload the resptr prop of '{link_prop.name}' of resource '{stashed_resource.id}'."
+            print(f"    WARNING: {err_msg} Original error message: {orig_err_msg}")
+            logger.warning(err_msg, exc_info=True)
+            continue
+        if verbose:
+            print(f'  Successfully uploaded resptr-prop of "{link_prop.name}". Value: {resptr}')
+            logger.info(f'Successfully uploaded resptr-prop of "{link_prop.name}". Value: {resptr}')
+        nonapplied_resptr_props[stashed_resource][link_prop].remove(resptr)
+    return nonapplied_resptr_props
+
+
+def _unable_to_retrieve_resource(
```

Comment:
Concerning the inline comment: They explain why after this function call, the loop is continued with the `continue` keyword. So it doesn't make sense to have the inline comment in the function. I would move it up, to where the function is called.

Reply:
Done.